lib/PathfinderUtil.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * PathfinderUtil.h, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+#pragma once
+
+#include "mapping/CMapDefines.h"
+#include "CGameState.h"
+
+namespace PathfinderUtil
+{
+	using FoW = std::vector<std::vector<std::vector<ui8> > >;
+	using ELayer = EPathfindingLayer;
+
+	template<EPathfindingLayer::EEPathfindingLayer layer>
```

Comment:
It is an optimization. There a `switch` on layer inside the function, because now layer is a constant inside it can be completely optimized out by compiler.